### PR TITLE
Add unit test for verifying the correct logger input is received in C++ client lib

### PR DESCRIPTION
**Pull request by jackgerrits**
Unit test to test for bug fixed in #1582

### CODE REVIEW

File: reinforcement_learning/unit_test/live_model_test.cc [hunk 1]
```diff
@@ -231,3 +232,87 @@ BOOST_AUTO_TEST_CASE(live_model_mocks) {
...
+
+BOOST_AUTO_TEST_CASE(live_model_logger_receive_data) {
+  Mock<r::i_logger> mock_observation_logger;
+  std::vector<std::string> recorded_observations;
+  When(Method(mock_observation_logger, init)).AlwaysReturn(err::success);
+  When(Method(mock_observation_logger, append)).AlwaysDo(
+    [&recorded_observations](std::string& message, reinforcement_learning::api_status* status) {
+    recorded_observations.push_back(message); return err::success;
+  });
+  Fake(Dtor(mock_observation_logger));
```
Comment (tyclintw) [inline reinforcement_learning/unit_test/live_model_test.cc]: nit: These mocks are pretty generic. In the future we should try to extract this from the unittest itself.

File: reinforcement_learning/unit_test/live_model_test.cc [hunk 2]
```diff
@@ -231,3 +232,87 @@ BOOST_AUTO_TEST_CASE(live_model_mocks) {
...
+  auto const num_iterations = 5;
+  for (auto i = 0; i < num_iterations; i++) {
+    BOOST_CHECK_EQUAL(model.choose_rank(event_id_1, JSON_CONTEXT, response), err::success);
+    BOOST_CHECK_EQUAL(model.report_outcome(event_id_1, 1.0), err::success);
+
+    BOOST_CHECK_EQUAL(model.choose_rank(event_id_2, JSON_CONTEXT, response), err::success);
+    BOOST_CHECK_EQUAL(model.report_outcome(event_id_2, 1.0), err::success);
+  }
+
+  auto const expected_interaction_1 = u::concat(R"({"Version":"1","EventId":")", event_id_1, R"(","a":[1,2],"c":)", JSON_CONTEXT, R"(,"p":[0.500000,0.500000],"VWState":{"m":"N/A"}})");
```
Comment (tyclintw) [inline reinforcement_learning/unit_test/live_model_test.cc]: >1 [](start = 63, length = 1)

I see "Version" is hard coded in ranking_event.cc but this should probably be dynamic. As the version increases the tests should hold.

Comment (jackgerrits) [reply reinforcement_learning/unit_test/live_model_test.cc]: Version is currently also hardcoded in the event serialization [see here](https://github.com/JohnLangford/vowpal_wabbit/blob/master/reinforcement_learning/rlclientlib/ranking_event.cc#L14). I agree we want to make it clear and easy to change this when the version does change.

File: reinforcement_learning/unit_test/live_model_test.cc (hunk 1 above)
Comment (jackgerrits) [reply reinforcement_learning/unit_test/live_model_test.cc]: I am going to leave this for another PR though in the interest of getting this one closed out

**Comment by jackgerrits** [comment]
I'll take a look at it today

**Comment by rajan-chari** [comment]
There are some conflicts with the refactored unit tests PR.  @jackgerrits  if you can resolve it, I can merge this.

**Pull request merged**